### PR TITLE
Request upload: test synchronous feature detect

### PR DESCRIPTION
**Pull request by jakearchibald**
Discussion: https://github.com/WebKit/standards-positions/issues/24#issuecomment-1181821440

### CODE REVIEW

**Review by lucacasonato — APPROVED** [review]
The code itself looks good to me, but I am not sure if this will actually have the intended outcome, as the implementer may just see this as "just another test failure for an unimplemented feature".

I can't think of anything better, and it definitely does not do any harm, so let's just land it.

**Review by annevk — APPROVED** [review]
Looks good to me modulo nit.

**Comment by annevk** [comment]
FWIW, I suspect as part of implementing Fetch features WebKit would sync `fetch/` tests and thus this would be noticed at the relevant time.

File: fetch/api/basic/request-upload.h2.any.js [hunk 1]
```diff
@@ -98,6 +102,63 @@ promise_test(async (test) => {
...
+
+  assert_equals(
+    request.headers.get("Content-Type"),
+    null,
+    `Request should not have a content-type set`
+  );
+  assert_true(duplexAccessed, `duplex dictionary property should be accessed`);
+}, "Synchronous feature detect");
+
+// The asserts the synchronousFeatureDetect isn't broken by a partial implementation.
```
[the remark this answers is not in the document]
Comment (annevk) [reply fetch/api/basic/request-upload.h2.any.js:125]: I forgot about this nit... Maybe fix in a follow-up?

**Pull request merged**